### PR TITLE
fix(work-item mapper): Added work-item mapper

### PR DESCRIPTION
**Pull request by divyanshiGupta**

### CODE REVIEW

File: src/app/models/work-item.ts [hunk 1]
```diff
@@ -91,11 +98,12 @@ export class LinkDict {
   count: number;
 }
 
+export interface WorkItemService extends WorkItem {}
 
 export interface WorkItemUI {
   id: string;
   title: string;
-  number: string;
+  sysNumber: string;
```
Comment (sanbornsen) [inline src/app/models/work-item.ts:106]: make it just ```number```

File: src/app/models/work-item.ts [hunk 2]
```diff
@@ -104,4 +112,81 @@ export interface WorkItemUI {
   createdAt: string;
   labels: LabelUI[];
   comments: CommentUI[];
+  wiState: string;
```
Comment (sanbornsen) [inline src/app/models/work-item.ts:115]: make it just ```state```

Comment (sanbornsen) [reply src/app/models/work-item.ts:115]: where is the mapper for this attribute?

File: src/app/models/work-item.ts [hunk 3]
```diff
@@ -104,4 +112,81 @@ export interface WorkItemUI {
...
+  wiDescription: string;
```
Comment (sanbornsen) [inline src/app/models/work-item.ts:116]: make it just ```description```

Comment (sanbornsen) [reply src/app/models/work-item.ts:116]: where is the mapper for this attribute?

File: src/app/models/work-item.ts [hunk 4]
```diff
@@ -104,4 +112,81 @@ export interface WorkItemUI {
...
+}
+
+export class WorkItemMapper implements Mapper<WorkItemService, WorkItemUI> {
+  itMapper = new IterationMapper();
+  wiTypeMapper = new WorkItemTypeMapper();
+  areaMapper = new AreaMapper();
+  userMapper = new UserMapper();
+  lMapper = new LabelMapper();
```
Comment (sanbornsen) [inline src/app/models/work-item.ts:124]: Why is it ```userMapper``` and ```areaMapper``` for them and just ```lMapper``` for ```label```? Same with the next one. Why is it ```cMapper``` not ```commentMapper```?

File: src/app/models/work-item.ts [hunk 5]
```diff
@@ -104,4 +112,81 @@ export interface WorkItemUI {
...
+  ];
+  
+  toUIModel(arg: WorkItemService): WorkItemUI {
+    let workItemUI = switchModel<WorkItemService, WorkItemUI>(
+      arg, this.serviceToUiMapTree
+    );
+    return this.tempToUIModel(arg, workItemUI);
+  }
+
+  tempToUIModel(arg: WorkItemService, workItemUI: WorkItemUI): WorkItemUI {
```
Comment (sanbornsen) [inline src/app/models/work-item.ts:164]: Why do we need these extra mapping functions for dependent attributes? Let's recap the flow once - 
* Action dispatch GET_WORKITEMS
* Effect listens for GET_WORKITEMS
* Effect maps it to UI model
* Effect resolves each the work items 
    - Takes the existing UI states of dependent attributes 
    - Pass them on to a function in this class along with one work item and resolve it
* Effect dispatches GET_WORKITEMS_SUCCESS

File: src/app/models/work-item.ts [hunk 6]
```diff
@@ -104,4 +112,81 @@ export interface WorkItemUI {
...
+  } 
+
+  toServiceModel(arg: WorkItemUI): WorkItemService {
+    let workItemService = switchModel<WorkItemUI, WorkItemService>(
+      arg, this.uiToServiceMapTree
+    );
+    return this.tempToServiceModel(arg, workItemService);
+  }
+
+  tempToServiceModel(arg: WorkItemUI, workItemService: WorkItemService): WorkItemService {
```
Comment (sanbornsen) [inline src/app/models/work-item.ts:182]: And this is also not needed for the same reason !!

**Review by sanbornsen — CHANGES_REQUESTED** [review]

File: src/app/models/work-item.ts (hunk 1 above)
Comment (divyanshiGupta) [reply src/app/models/work-item.ts:106]: but isnt sysNumber more descriptive about the attribute? as we are getting system.number in this particular attribute.

**Pull request closed without merging**